File: Nordax.Bank.Recruitment/Models/LoanApplication/RegisterLoanApplicationRequest.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Nordax.Bank.Recruitment.Models.LoanApplication
{
	public class RegisterLoanApplicationDocumentRequest
	{
		[Required] public string DocumentType { get; set; }

		[Required] public string FileRef { get; set; }
	}

	public class RegisterLoanApplicationRequest
	{
		[Required] public string ApplicantOrganizationNo { get; set; }

		[Required] public string ApplicantFirstName { get; set; }

		[Required] public string ApplicantSurname { get; set; }

		[Required] public string ApplicantPhoneNo { get; set; }

		[Required] public string ApplicantEmail { get; set; }

		public string ApplicantAddress { get; set; }

		public string ApplicantIncomeLevel { get; set; }

		public bool ApplicantIsPoliticallyExposed { get; set; }

		public decimal LoanAmount { get; set; }

		public int LoanPaymentPeriod { get; set; }

		public int LoanBindingPeriod { get; set; }

		public decimal LoanInterestRate { get; set; }

		[Required] public List<RegisterLoanApplicationDocumentRequest> Documents { get; set; }
	}
}

File: Nordax.Bank.Recruitment/ClientApp/src/models/registerLoanApplicationRequest.ts
import { hasIndexer } from "../common/classUtil";

export interface RegisterLoanApplicationRequest extends hasIndexer {
    applicantOrganizationNo: string;
    applicantFirstName: string;
    applicantSurname: string;
    applicantPhoneNo: string;
    applicantEmail: string;
    applicantAddress: string;
    applicantIncomeLevel: string;
    applicantIsPoliticallyExposed: boolean;

    loanAmount: number;
    loanPaymentPeriod: number;
    loanBindingPeriod: number;
    loanInterestRate: number;

    documents: RegisterLoanApplicationDocumentRequest[];
}

export interface RegisterLoanApplicationDocumentRequest {
    documentType: string;
    fileRef: string;
}